src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2193,23 +2196,37 @@ void TorrentImpl::handleSaveResumeDataAlert(const lt::save_resume_data_alert *p)
 
 void TorrentImpl::prepareResumeData(const lt::add_torrent_params &params)
 {
-    if (m_hasMissingFiles)
     {
```

Comment:
Is it intentional to leave a scope here? just double checking.

Reply:
It is done to intentionally end the scope of variables immediately after they are moved.